pkg/subctl/cmd/gather/servicediscovery.go

```diff
@@ -110,10 +114,31 @@ func gatherConfigMapCoreDNS(info *Info) {
 	}
 }
 
+// gatherInternalServices gathers the internal service created for every exported submariner service.
+func gatherInternalServices(info *Info) {
+	services, err := info.ClientProducer.ForKubernetes().CoreV1().Services(corev1.NamespaceAll).List(context.TODO(),
+		metav1.ListOptions{LabelSelector: internalSvcLabel})
+	if err != nil {
+		info.Status.Warning("Error gathering the internal service configuration: %v", err)
+	}
+
+	if len(services.Items) > 0 {
+		gatherInternalService(info, corev1.NamespaceAll, internalSvcLabel)
+	}
```

Comment:
The `List` call is extraneous - this can simply be reduced to call `ResourcesToYAMLFile`.

```suggestion
	ResourcesToYAMLFile(info, schema.GroupVersionResource{
		Group:    corev1.SchemeGroupVersion.Group,
		Version:  corev1.SchemeGroupVersion.Version,
		Resource: "services",
	}, corev1.NamespaceAll, metav1.ListOptions{LabelSelector: internalSvcLabel})
```

Reply:
Since it could be reduced down so much I went ahead and made it more generic `gatherLabeledServices` for reuse in future gathers as needed. Ty for taking a look!